packages/nimble-components/src/patterns/button/styles.ts

```diff
@@ -49,7 +50,7 @@ export const styles = css`
         background-color: transparent;
         height: 100%;
         width: 100%;
-        border: ${borderWidth} solid transparent;
+        border: 0px solid transparent;
```

Comment:
Just gonna capture some thoughts here:

## Control sizing

This is a change in how we do the sizing of components. For example see the following two screenshots in devtools.

main:
![image](https://github.com/ni/nimble/assets/1588923/c979582b-f0d2-4f22-8d3e-7eef491cdbbc)

PR:
![image](https://github.com/ni/nimble/assets/1588923/efc31011-1e4c-4bf7-95cb-08b3993e1d6a)

So for example in those screenshots the definition of how the 16px green padding is applied changes. It becomes 16px to the outer edge of the control instead of 16px to the start of the border.

- Seems like this should be done consistently across all controls otherwise reasoning about sizing becomes difficult / we might end up with increasing inconsistencies
- The "border" draws inset to the control instead of outset. That seems reasonable I think.
  - Is that what figma does for measuring sizes as well? I found [one article](https://threads.com/blog/34432545238/why-we-use-the-css-property-boxshadow-instead-of-border-in-our-app) that seems to indicate borders are outset and don't contribute to size instead of inset. It may be useful to know how figma vs xd handle that. Maybe that could lean us to preferring an outset?

## Performance

This is adding additional box-shadow drawings to the button and based on the above potentially to every control. At least historically drawing box-shadows was pretty expensive. I'm having difficulty finding good references for performance today but one thing that does seem consistent is that animating box shadows is expensive.

- Can we characterize the performance cost of using box-shadows for all borders? Mobile / low-end devices performance ?
- Can we use an approach [to animate opacity instead](https://css-tricks.com/getting-deep-into-shadows/#aa-animating-shadows) of two overlaid box shadows instead of animating the the box-shadow itself?

Reply:
I talked with Molly about the benefits and drawbacks of using the additional box-shadows yesterday, and I think I'm going to revert it back to the normal border styling that we use in main. Based on what you mentioned regarding styling differences and possible performance costs, this issue with box-shadow where the text overflows beyond the background-color (see screenshot), and my discussion with Molly, I don't think it's worth it implementing box-shadow to fix the small issue of a tiny white between the two border pixels that's only visible when zooming in 4-6 times in storybook.

![boxshadowbutton](https://github.com/ni/nimble/assets/103057880/78e09b60-4eeb-4d59-a851-b7da4e737fcc)

I'm working on getting this code updated to reflect this now, and I'll show the diffs in the team meeting today.